C24_WMDE_Desktop_DE_19

Sperling-0 opened the pull request:
- `Both banners`:

  - The banner is based on the VAR banner of desktop-de-18
 
  - Emit the event `modalOpened` when the use of funds modal is opened.

  - Emit the event `modalClosed` when the use of funds modal is closed.
  
 - `Var`:

   - The variant banner design is changed according to the design file for the [slider](https://www.figma.com/design/GZ1lU5L8mpaLXvtwlNfMki/wikimedia-FUNDRAISING?node-id=3078-404&t=0TCV5fzWZENGGqLh-4) and the [non-slider](https://www.figma.com/design/GZ1lU5L8mpaLXvtwlNfMki/wikimedia-FUNDRAISING?node-id=3078-409&t=Cg3T7nNprwjxCnx8-4).
   
   - The variant banner integrates the "10 good reasons" feature using the sticker.
  
   - Diverging from the design, the additional element should be at the bottom (sharing the space with the slide indicators) to prevent overlapping.

   - Hovering over the new element
       -  turns the cursor into a pointer.
       -  adds underlining to the link text.

   - Interaction with `ReasonsToDonate` is tracked.

Ticket: https://phabricator.wikimedia.org/T380852

Sperling-0 committed:
Prepare campaign for C24_WMDE_Desktop_DE_19

Ticket: https://phabricator.wikimedia.org/T380852

Sperling-0 committed:
Prepare CTRL for C24_WMDE_Desktop_DE_19

- The banner is based on the `VAR` banner of desktop-de-18
- Emit the event `modalOpened` when the use of funds modal is opened
- Emit the event `modalClosed` when the use of funds modal is closed
- Add fallback banner slider colors to Treedip - we might change that in
  the future

Ticket: https://phabricator.wikimedia.org/T380852

Sperling-0 committed:
Prepare VAR for C24_WMDE_Desktop_DE_19

Ticket: https://phabricator.wikimedia.org/T380852

Sperling-0 committed:
Implement VAR for C24_WMDE_Desktop_DE_19

- Emit the event `modalOpened` when the use of funds modal is opened.
- Emit the event `modalClosed` when the use of funds modal is closed.
- The variant banner design is changed according to the design file for the slider and the non-slider.
- The variant banner integrates the "10 good reasons" feature using the sticker.
- Diverging from the design, the additional element should be at the bottom (sharing the space with the slide indicators) to prevent overlapping.
- Hovering over the new element:
   - turns the cursor into a pointer.
   - adds underlining to the link text.
- Interaction with `ReasonsToDonate` is tracked
- Add tests for VAR banner

Ticket: https://phabricator.wikimedia.org/T380852

moiikana commented:
there are no tests yet for the var feature (reasons to donate).
you could probably use almost the same tests as in desktop 14

moiikana committed:
Prepare C24_WMDE_Desktop_DE_19 for lateprogress

- enable lateprogress in the local dev environment
- use ProgressBarAlternative to show donation goal to the right of the bar

- use ProgressBarAlternative in FallbackBanner as well

https://phabricator.wikimedia.org/T381957

moiikana commented:
we could give `.wmde-banner-10-reasons-close button` a `cursor: pointer;` to help the user understanding that the X is a button

Sperling-0 committed:
C24_WMDE_Desktop_DE_19 - Improve reasons to donate modal close button

- Add curson pointer to the close button

Ticket: https://phabricator.wikimedia.org/T380852

moiikana commented on test/features/ReasonsToDonate.ts:
```diff
@@ -21,23 +21,24 @@ const expectScrollsToFormWhenCallToActionIsClicked = async ( wrapper: VueWrapper
 	await wrapper.find( '.wmde-banner-10-reasons-cta button' ).trigger( 'click' );
 
 	expect( tracker.trackEvent ).toHaveBeenCalledTimes( 2 );
+	// expect( tracker.trackEvent ).toBeCalledWith( new ReasonsToDonateShownEvent() );
```
there is a leftover code commented out here

moiikana commented on test/features/ReasonsToDonate.ts:
```diff
@@ -12,7 +12,7 @@ const expectContainsReasonsToDonateDialogue = async ( wrapper: VueWrapper<any> )
 
 const expectScrollsToFormWhenCallToActionIsClicked = async ( wrapper: VueWrapper<any>, pageScroller: PageScroller ): Promise<any> => {
-	await wrapper.find( '.wmde-banner-reasons-to-donate-link' ).trigger( 'click' );
+	await wrapper.find( '.ten-good-reasons-sticker-text' ).trigger( 'click' );
```
changing the tests like this will make the Desktop DE 14 and Mobile DE 09 tests fail because they rely on them.

Maybe you can duplicate the tests and have "sticker"  vs. "link" as a click source (for the naming of the tests)

Sperling-0 replied:
Thank you for pointing it out. I added new tests for 'sticker'.
I did not rename old tests to be 'link' tests, because then `desktop-de-14` and `mob-de-09` would have to be changed to reflect the rename changes.
